src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementServiceTest.java

```diff
@@ -0,0 +1,383 @@
+            camundaService,
+            camundaQueryBuilder,
+            permissionEvaluatorService,
+            cftTaskDatabaseService,
+            cftTaskMapper
+        );
+        mockServices.mockServiceAPIs();
+    }
+
+    protected Map<String, CamundaVariable> createMockCamundaVariables() {
+
+        Map<String, CamundaVariable> variables = new HashMap<>();
+        variables.put("caseId", new CamundaVariable("00000", "String"));
+        variables.put("caseName", new CamundaVariable("someCaseName", "String"));
+        variables.put("caseTypeId", new CamundaVariable("someCaseType", "String"));
+        variables.put("taskState", new CamundaVariable("configured", "String"));
+        variables.put("location", new CamundaVariable("someStaffLocationId", "String"));
+        variables.put("locationName", new CamundaVariable("someStaffLocationName", "String"));
+        variables.put("securityClassification", new CamundaVariable("SC", "String"));
+        variables.put("title", new CamundaVariable("some_title", "String"));
+        variables.put("executionType", new CamundaVariable("some_executionType", "String"));
+        variables.put("taskSystem", new CamundaVariable("some_taskSystem", "String"));
+        variables.put("jurisdiction", new CamundaVariable("some_jurisdiction", "String"));
+        variables.put("region", new CamundaVariable("some_region", "String"));
+        variables.put("appealType", new CamundaVariable("some_appealType", "String"));
+        variables.put("autoAssigned", new CamundaVariable("false", "Boolean"));
+        variables.put("assignee", new CamundaVariable("uid", "String"));
+        return variables;
+    }
+
+    protected CamundaTask createMockedUnmappedTask() {
+        return new CamundaTask(
+            "someCamundaTaskId",
+            "someCamundaTaskName",
+            IDAM_USER_ID,
+            ZonedDateTime.now(),
+            ZonedDateTime.now().plusDays(1),
+            "someCamundaTaskDescription",
+            "someCamundaTaskOwner",
+            "someCamundaTaskFormKey",
+            "someProcessInstanceId"
+        );
+    }
+
+    private void verifyTransactionWasRolledBack(String taskId) {
+
+        //Find the task
+        Optional<TaskResource> savedTaskResource = taskResourceRepository.findById(taskId);
+
+        assertNotNull(savedTaskResource);
+        assertTrue(savedTaskResource.isPresent());
+        assertEquals(taskId, savedTaskResource.get().getTaskId());
+        assertEquals("taskName", savedTaskResource.get().getTaskName());
+        assertEquals("taskType", savedTaskResource.get().getTaskType());
+
+        //Because transaction was rolled back we should have no state
+        assertEquals(null, savedTaskResource.get().getState());
+
+    }
+
+    private void createAndSaveTestTask(String taskId) {
+        TaskResource taskResource = new TaskResource(taskId, "taskName", "taskType");
+        taskResourceRepository.save(taskResource);
+    }
+
+    @Nested
```

Comment:
I like the nested thing to organise it better and reuse stuff from the base class. We could also move these nested classes to their own class and extend from this base class if we see the file gets too big over time.